src/server/editorServices.ts

```diff
@@ -1538,13 +1819,11 @@ namespace ts.server {
                         const change = file.changes[i];
                         scriptInfo.editContent(change.span.start, change.span.start + change.span.length, change.newText);
                     }
-                    if (recordChangedFiles) {
-                        if (!this.changedFiles) {
-                            this.changedFiles = [scriptInfo];
-                        }
-                        else if (this.changedFiles.indexOf(scriptInfo) < 0) {
-                            this.changedFiles.push(scriptInfo);
-                        }
+                    if (!this.changedFiles) {

```

Comment:
This would make a good helper function, `this.changedFiles = addIfNotPresent(this.changedFiles, scriptInfo)`.